lms/static/scripts/frontend_apps/services/client-rpc.js

```diff
@@ -24,17 +59,37 @@ export class ClientRpc {
    * @param {Object} options
    *   @param {string[]} options.allowedOrigins -
    *     Origins that are allowed to request client configuration
-   *   @param {Object} options.clientConfig -
+   *   @param {string} options.authToken -
+   *     Auth token used in LMS backend API calls to refresh the grant token
+   *     in the `clientConfig` if it has expired
+   *   @param {ClientConfig} options.clientConfig -
    *     Configuration for the Hypothesis client. Whatever is provided here is
    *     passed directly to the client via `window.postMessage` when it requests
    *     configuration. It should be a subset of the config options specified at
    *     https://h.readthedocs.io/projects/client/en/latest/publishers/config/.
    */
-  constructor({ allowedOrigins, clientConfig }) {
+  constructor({ allowedOrigins, authToken, clientConfig }) {
     this._server = new Server(allowedOrigins);
 
-    // Handle the initial request for configuration from the Hypothesis client.
-    this._server.register('requestConfig', () => clientConfig);
+    // Handle the requests for configuration from the Hypothesis client.
+    this._server.register('requestConfig', async () => {
+      // Refresh the grant token used by the client to access the H API, if
+      // it expired.
+      const grantToken = clientConfig.services[0].grantToken;
+      const now = Date.now();
+      if (now > jwtExpiry(grantToken)) {
+        const response = await apiCall({
+          authToken,
+          path: '/api/grant_token',
+        });
+        if (typeof response.grant_token !== 'string') {
```

Comment:
Can you help me understand in what situations there could be a response body but the `grant_token` is non-string?